app/forms/candidate_interface/english_gcse_grade_form.rb

```diff
@@ -144,20 +144,20 @@ def assign_values(params)
       self
     end
 
-    def save_grades
-      if !valid?(:structured_grades)
-        log_validation_errors(:structured_grades)
-        return false
-      end
-      qualification.update(structured_grades: build_grades_json, grade: nil)
-    end
-
-    def save_grade
-      if !valid?(:grade)
-        log_validation_errors(:grade)
-        return false
+    def save
+      if is_multiple_gcse?
+        if !valid?(:structured_grades)
```

Comment:
Worth extracting a little more for reaability?
You could have something like

```
def save
  is_multiple_gcse? ? save_grades : save_grade
end

```

Reply:
Good idea, thanks